runpod_worker/client_example.py

```
@@ -0,0 +1,283 @@
+                    raise Exception(f"Training failed: {error}")
+                elif status in ['IN_QUEUE', 'IN_PROGRESS']:
+                    elapsed_minutes = (attempt + 1) * 0.5
+                    print(f"⏳ Training in progress... {elapsed_minutes:.1f} minutes elapsed")
+                    time.sleep(30)  # Wait 30 seconds
+                else:
+                    print(f"🔄 Status: {status}")
+                    time.sleep(30)
+                    
+            except requests.RequestException as e:
+                if attempt == max_attempts - 1:
+                    raise Exception(f"Failed to check job status: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)